nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITListAzureBlobStorage.java

```diff
@@ -55,12 +55,58 @@ public void testListBlobsUsingCredentialService() throws Exception {
         assertResult();
     }
 
+    @Test
+    public void testListWithMinAge() throws Exception {
+        runner.setProperty(ListAzureBlobStorage.MIN_AGE, "1 hour");
+
+        runner.assertValid();
+        runner.run(1);
+
+        runner.assertTransferCount(ListAzureBlobStorage.REL_SUCCESS, 0);
+    }
+
+    @Test
+    public void testListWithMaxAge() throws Exception {
+        runner.setProperty(ListAzureBlobStorage.MAX_AGE, "1 hour");
+
+        runner.assertValid();
+        runner.run(1);
+
+        assertResult(TEST_FILE_CONTENT);
+    }
+
+    @Test
+    public void testListWithMinSize() throws Exception {
+        uploadTestBlob("nifi-test-blob2", "Test");
+        runner.setProperty(ListAzureBlobStorage.MIN_SIZE, "5 B");
+
+        runner.assertValid();
+        runner.run(1);
+
+        assertResult(TEST_FILE_CONTENT);
+    }
+
+    @Test
+    public void testListWithMaxSize() throws Exception {
+        uploadTestBlob("nifi-test-blob2", "Test");
+        runner.setProperty(ListAzureBlobStorage.MAX_SIZE, "5 B");
+
+        runner.assertValid();
+        runner.run(1);
+
+        assertResult("Test");
+    }
+
     private void assertResult() {
+        assertResult(TEST_FILE_CONTENT);
+    }
+
+    private void assertResult(final String contentLengthAsString) {
         runner.assertTransferCount(ListAzureBlobStorage.REL_SUCCESS, 1);
         runner.assertAllFlowFilesTransferred(ListAzureBlobStorage.REL_SUCCESS, 1);
 
         for (MockFlowFile entry : runner.getFlowFilesForRelationship(ListAzureBlobStorage.REL_SUCCESS)) {
-            entry.assertAttributeEquals("azure.length", "36");
+            entry.assertAttributeEquals("azure.length", String.valueOf(contentLengthAsString.length()));
```

Comment:
The test uploads the test string converted to bytes (`fileContent.getBytes()`) and then the length of the binary payload is written to the attribute. Asserting to the length of the string may cause test failure in case of national characters. It is not an issue currently.